pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -373,7 +373,7 @@ func fixSystemdUnit(unitName string, unitContent string, shouldBeRunning bool) e
 	running := systemdUnitRunning(sd, unitName)
 	if !running && shouldBeRunning {
 		logging.Debugf("Starting %s", unitName)
-		return sd.Start(unitName)
+		return sd.EnableAndStart(unitName)
```

Comment:
This will also enable `crc-daemon.service`, but this service is socket activated, so should not be enabled.
Is this working without additional changes? When I looked into this, I had to add this to the unit files I wanted to enable:
```
[Install]
WantedBy=default.target
```

Reply:
@cfergeau yes I added that in one of the commit.

Reply:
> This will also enable crc-daemon.service, but this service is socket activated, so should not be enabled.

Never mind, this is wrapped in `if shouldBeRunning` which is there for this purpose, to differentiate between the `socket` units and the `service` unit.
One note, on upgrades this won't enable the socket units if they are running already. When the user hits the bug and reruns `setup`, then they will be upgraded as expected.